bootloader/bootloader.go

```diff
@@ -357,12 +366,11 @@ func ForGadget(gadgetDir, rootDir string, opts *Options) (Bootloader, error) {
 	if forcedBootloader != nil || forcedError != nil {
 		return forcedBootloader, forcedError
 	}
-	for _, blNew := range bootloaders {
-		bl := blNew(rootDir, opts)
-		markerConf := filepath.Join(gadgetDir, bl.Name()+".conf")
+	for name, blNew := range bootloaders {
+		markerConf := filepath.Join(gadgetDir, name+".conf")
```

Comment:
Hm I think I'd rather keep the previous way of creating the bootloader and then checking for existence of `bl.Name()+".conf"`, unless there's a reason to reverse the order of like it is now.

Reply:
Thinking a bit more, what if we cold do:
```
blName, err := gadget.RequiredBootloader(gadgetDir)
..
blNew := bootloaders[blName]
..
bl := blNew(rootDir, opts)
bl.InstallableFor(gadgetDir) // checks presence of Name()+".conf" internally
```

Reply:
opened this for the gadget: https://github.com/snapcore/snapd/pull/9677

Reply:
this is now the old way of doing things again